tests/healthReadyTests.js

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2021-2022 Inclusive Design Research Centre, OCAD University
+ * All rights reserved.
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ */
+
+"use strict";
+
+require("json5/lib/register");
+const fluid = require("infusion");
+const jqUnit = require("node-jqunit");
+
+require("../src/shared/driverUtils.js");
+require("./shared/utilsCommon.js");
+require("./shared/utilsSso.js");
+
+jqUnit.module("Personal Data Server /health and /ready tests.");
+
+fluid.registerNamespace("fluid.tests.healthReady");
+
+const skipDocker = process.env.SKIPDOCKER === "true" ? true : false;
+const path = require("path");
+const config = require("../src/shared/utils.js").loadConfig(path.join(__dirname, "testConfig.json5"));
+const serverUrl = "http://localhost:" + config.server.port;
+fluid.tests.utils.setDbEnvVars(config.db);
+
+const postgresOps = require("../src/dbOps/postgresOps.js");
+const postgresHandler = new postgresOps.postgresOps(config.db);
+
+const server = require("../server.js");
+
+jqUnit.test("Health and Ready end point tests", async function () {
+    jqUnit.expect(skipDocker ? 16 : 18);
+    let serverStatus, response;
+
+    // Start with server off -- "/health" should fail
+    response = await fluid.tests.utils.sendRequest(serverUrl, "/health");
+    jqUnit.assertNotNull("Check '/health' error", response);
+    jqUnit.assertTrue("Check '/health' error code", response.toString().includes("ECONNREFUSED"));
```

Comment:
This test fails for me in my Ubuntu image - I get "OK" instead

Reply:
> This test fails for me in my Ubuntu image - I get "OK" instead

At the end of every SSO test, there's a call to `server.stopServer()` to stop the server. As you hit a couple of test failures, I suspect it might be the previous test didn't reach the stopServer call so the server was still running.

Can you use command `lsof -i tcp:3001` to check if 3001 is open? If it is, kill the process. Then re-run the test to see how it goes. Thanks.

Reply:
Hi - I tried the lsof line which returned nothing but there was indeed an existing server running on port 3001. In this case it's puzzling that the tests run at all - they should terminate in an error at starting the server whereas it looks like at present they ignore the error and continue:

````
Finding the server running status at http://localhost:3001
... attempt 0
Port 3001 is already in use
````